includes/class-wc-order-item.php

```diff
@@ -0,0 +1,506 @@
+ * @since       2.6.0
+ * @package		WooCommerce/Classes
+ * @author 		WooThemes
+ */
+class WC_Order_Item implements ArrayAccess, WC_Data {
+
+    /**
+	 * Data array, with defaults.
+	 * @since 2.6.0
+	 * @var array
+	 */
+    protected $_data = array(
+		'order_id'      => 0,
+		'order_item_id' => 0,
+		'name'          => '',
+		'type'          => '',
+	);
+
+    /**
+     * Stores additonal meta data.
+     * @var array
+     */
+    protected $_meta_data = array();
+
+    /**
+	 * Constructor.
+	 * @param int|object|array $order_item ID to load from the DB (optional) or already queried data.
+	 */
+    public function __construct( $item = 0 ) {
+		if ( $item instanceof WC_Order_Item ) {
+            if ( $this->is_type( $item->get_type() ) ) {
+                $this->set_all( $item->get_data() );
+            }
+        } elseif ( is_array( $item ) ) {
+            $this->set_all( $item );
+		} else {
+            $this->read( $item );
+        }
+    }
+
+    /**
+     * Set all data based on input array.
+     * @param array $data
+     * @access private
+     */
+    public function set_all( $data ) {
+        foreach ( $data as $key => $value ) {
+            if ( is_callable( array( $this, "set_$key" ) ) ) {
+                $this->{"set_$key"}( $value );
+            } elseif ( 'meta_data' !== $key ) {
+                $this->_data[ $key ] = $value;
+            } else {
+                foreach ( $value as $meta_id => $meta ) {
+                    $this->_meta_data[ $meta_id ] = $meta;
+                }
+            }
+        }
+    }
+
+    /**
+     * Change data to JSON format.
+     * @return string Data in JSON format.
+     */
+    public function __toString() {
+        return json_encode( $this->get_data() );
+    }
+
+    /**
+     * Type checking
+     * @param  string|array  $Type
+     * @return boolean
+     */
+    public function is_type( $type ) {
+        return is_array( $type ) ? in_array( $this->get_type(), $type ) : $type === $this->get_type();
+    }
+
+    /**
+     * Get qty.
+     * @return int
+     */
+    public function get_qty() {
+        return 1;
+    }
+
+    /*
+	|--------------------------------------------------------------------------
+	| Getters
+	|--------------------------------------------------------------------------
+	*/
+
+    /**
+	 * Get all class data in array format.
+	 * @since 2.6.0
+	 * @return array
+	 */
+	public function get_data() {
+		return array_merge( $this->_data, array( 'meta_data' => $this->get_meta_data() ) );
+	}
+
+    /**
+     * Get order item ID.
+     * @return int
+     */
+    public function get_id() {
+        return $this->get_order_item_id();
+    }
+
+    /**
+     * Get order ID this meta belongs to.
+     * @return int
+     */
+    public function get_order_id() {
+        return absint( $this->_data['order_id'] );
+    }
+
+    /**
+     * Get order ID this meta belongs to.
+     * @return int
+     */
+    public function get_order_item_id() {
```

Comment:
order_item_id is part of _data so this was added for consistency with other _data props.